opendatahub-io/odh-dashboard
# RHOAIENG-4346 Harden backend security

Closes: [RHOAIENG-4346](https://issues.redhat.com//browse/RHOAIENG-4346)

## Description


Removes the backend `/api/proxy` endpoint and replaces it with bespoke service endpoints for pipelines, trustyai and mlmd apis.

## How Has This Been Tested?



- Deploy to a cluster
- Verify pipelines functionality works as expected
- Verify bias metrics functionality works as expected

## Test Impact


Existing tests updated and reworked to cover changes to functionality.
## Request review criteria:



Self checklist (all need to be checked):

If you have UI changes: 


After the PR is posted & before it merges:
